Unity-Technologies/ScriptableRenderPipeline
# Fix for decal system mip streaming texture properties update

### Purpose of this PR
Decal system now only updates texture mip streaming settings when using expected non shader graph material. This fixes a test failure assert that was missed in the original PR.
When material is no longer used the texture mip streaming settings are reset.

Original PR
https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/3755

---
### Release Notes

---
### Testing status
**Katana Tests**: 
Ran and properly checked results. All good.
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=decals-texture-streaming-fix-update&automation-tools_branch=master&unity_branch=trunk&sort=4-asc

**Yamato Tests**: 
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/decals-texture-streaming-fix-update/.yamato%252Fupm-ci-projects.yml%2523all_project_ci/156469/job/pipeline

One known issue with PostProcessingGraphicsTests.0205_Std_Grain on Test PostProcessing on Windows64_Standalone
(Confirmed with Sophia Clarke)

**Manual Tests**: What did you do?
As original PR
- Created a decal with texture containing mips and set to streaming.
- Enabled texture streaming and set max mip level reduction high (to 7)
- Prior to this fix the texture was limited to a low mip so looks blurry.
- With this fix all the mips are loaded and is no longer blurry.
There is also a test in the bug report listed above.

+ Checked the previous failing test in the HDRP test suite (4001-expose) now passes (along with others)

**Automated Tests**: 
None setup

---
### Overall Product Risks
**Technical Risk**: Low
**Halo Effect**: Low

---
### Comments to reviewers
Katana passed now
